arangod/Cluster/Maintenance.cpp

```diff
@@ -0,0 +1,1052 @@
+    if (collection == nullptr) {
+      std::string errorMsg(
+        "Maintenance::assembleLocalCollectionInfo: Failed to lookup collection ");
+      errorMsg += shard;
+      LOG_TOPIC(DEBUG, Logger::MAINTENANCE) << errorMsg;
+      { VPackObjectBuilder o(&ret); }
+      return ret;
+    }
+
+    { VPackObjectBuilder r(&ret);
+      auto it = allErrors.shards.find(
+          database + "/" + std::to_string(collection->planId()) + "/" + shard);
+      if (it == allErrors.shards.end()) {
+        ret.add(ERROR, VPackValue(false));
+        ret.add(ERROR_MESSAGE, VPackValue(std::string()));
+        ret.add(ERROR_NUM, VPackValue(0));
+      } else {
+        VPackSlice errs(static_cast<uint8_t const*>(it->second->data()));
+        ret.add(ERROR, errs.get(ERROR));
+        ret.add(ERROR_NUM, errs.get(ERROR_NUM));
+        ret.add(ERROR_MESSAGE, errs.get(ERROR_MESSAGE));
+      }
+      ret.add(VPackValue(INDEXES));
+      { VPackArrayBuilder ixs(&ret);
+        if (info.get(INDEXES).isArray()) {
+          auto it1 = allErrors.indexes.find(shard);
+          std::unordered_set<std::string> indexesDone;
+          // First the indexes as they are in Local, potentially replaced
+          // by an error:
+          for (auto const& index : VPackArrayIterator(info.get(INDEXES))) {
+            std::string id = index.get(ID).copyString();
+            indexesDone.insert(id);
+            if (it1 != allErrors.indexes.end()) {
+              auto it2 = it1->second.find(id);
+              if (it2 != it1->second.end()) {
+                // Add the error instead:
+                ret.add(VPackSlice(
+                    static_cast<uint8_t const*>(it2->second->data())));
+                continue;
+              }
+            }
+            ret.add(removeSelectivityEstimate(index).slice());
+          }
+          // Now all the errors for this shard, for which there is no index:
+          if (it1 != allErrors.indexes.end()) {
+            for (auto const& p : it1->second) {
+              if (indexesDone.find(p.first) == indexesDone.end()) {
+                ret.add(VPackSlice(
+                      static_cast<uint8_t const*>(p.second->data())));
+              }
+            }
+          }
+        }
+      }
+      ret.add(VPackValue(SERVERS));
+      { VPackArrayBuilder a(&ret);
+        ret.add(VPackValue(ourselves));
+        auto current = *(collection->followers()->get());
+        for (auto const& server : VPackArrayIterator(planServers)) {
+          if (std::find(current.begin(), current.end(), server.copyString())
+              != current.end()) {
+            ret.add(server);
+          }
+        }}}
+    
+    return ret;
+  } catch (std::exception const& e) { 
+    std::string errorMsg(
+      "Maintenance::assembleLocalCollectionInfo: Failed to lookup database ");
+    errorMsg += database;
+    errorMsg += " exception: ";
+    errorMsg += e.what();
+    LOG_TOPIC(DEBUG, Logger::MAINTENANCE) << errorMsg;
```

Comment:
An exception is thrown.  Sounds like an ERR log message to me, not DEBUG.  DEBUG translates to "hidden".